velox/functions/sparksql/tests/ComparisonsTest.cpp

```diff
@@ -86,9 +134,67 @@ TEST_F(ComparisonsTest, equaltonullsafe) {
   EXPECT_EQ(equaltonullsafe<double>(kNaN, std::nullopt), false);
   EXPECT_EQ(equaltonullsafe<double>(kNaN, 1), false);
   EXPECT_EQ(equaltonullsafe<double>(kNaN, kNaN), true);
+
+  assertArrayInput("null", "null", true);
+  assertArrayInput("null", "[1.0]", false);
+  assertArrayInput("[1.0]", "null", false);
+
+  assertArrayInput("[]", "[]", true);
+
+  assertArrayInput("[1.0, 2.0, 3.0]", "[1.0, 2.0, 3.0]", true);
+  assertArrayInput("[1.0, 2.0, 3.0]", "[1.0, 2.0, 4.0]", false);
+
+  assertArrayInput("[1.0, null]", "[6.0, 2.0]", false);
+  assertArrayInput("[1.0, null]", "[1.0, 2.0]", false);
+  assertArrayInput("[1.0, NaN]", "[1.0, NaN]", true);
+
+  assertArrayInput("[]", "[null, null]", false);
+  assertArrayInput("[1.0, 2.0]", "[1.0, 2.0, null]", false);
+  assertArrayInput("[null, null]", "[null, null, null]", false);
+
+  assertArrayInput("[null, null]", "[null, null]", true);
+
+  assertRowInput(std::nullopt, std::nullopt, true);
+  assertRowInput(std::nullopt, 1, false);
+  assertRowInput(1, std::nullopt, false);
+
+  assertRowInput(1, 2, false);
+  assertRowInput(1, 1, true);
+
+  runAndCompareForRowInput(
```

Comment:
Is any reason for not using assertRowInput? Thanks.

Reply:
This functions is for testing null row case, for example, comparing row(1) VS. null.

Reply:
Got it. Thanks.